gpt4all-bindings/java/src/main/java/com/hexadevlabs/gpt4all/LLModel.java

```diff
@@ -184,11 +184,16 @@ public LLModel(Path modelPath) {
             throw new IllegalStateException("Model file does not exist: " + modelPathAbs);
         }
 
+        // Check if file is Readable
+        if(Files.isReadable(modelPath)){
+            throw new IllegalStateException("Model file cannot be read: " + modelPathAbs);
```

Comment:
Shouldn't you check if file is not readable and then throw exception?

Reply:
Yes definatly!